### PR TITLE
RFR: Update README for unit/functional testing

### PR DESCRIPTION
**Pull request by bsquizz**
Improve our documentation related to running the unit/functional tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,19 +77,24 @@ The diagram below describes how the application accesses the configuration.
...
-enables the operator to get initialized against a control pane just like it
-would against a real Openshift cluster.
+The tests rely on the test environment set up by controller-runtime.  This enables the operator to 
+get initialized against a control pane just like it would against a real Openshift cluster.
+
+1. Download and install `kubebuilder`.
+[See install instructions here](https://book.kubebuilder.io/quick-start.html#installation).
+2. Make sure that `kubebuilder` is present in your `PATH`. It is recommended you append it to your
+`PATH` in your `.bashrc`, `.zshrc`, or similar.
+3. Download and install `kustomize`.
```
Comment (psav) [inline README.md:87]: Kustomize isn't required for the current tests - is this required for the new unit tests you added?

Comment (bsquizz) [reply README.md:87]: I was getting some errors about kustomize not being in my PATH when I ran the tests initially :man_shrugging:

Comment (kylape) [reply README.md:87]: I think you need kustomize to do the initial code generation done by the SDK.  Worst case devs install kustomize when they don't have to, which doesn't seem that bad.

File: README.md [hunk 2]
```diff
@@ -77,19 +77,24 @@ The diagram below describes how the application accesses the configuration.
...
 
-The tests rely on the test environment set up by controller-runtime.  This
-enables the operator to get initialized against a control pane just like it
-would against a real Openshift cluster.
+The tests rely on the test environment set up by controller-runtime.  This enables the operator to 
+get initialized against a control pane just like it would against a real Openshift cluster.
+
+1. Download and install `kubebuilder`.
+[See install instructions here](https://book.kubebuilder.io/quick-start.html#installation).
+2. Make sure that `kubebuilder` is present in your `PATH`. It is recommended you append it to your
```
Comment (kylape) [inline README.md:85]: Just putting the kubebuilder binary in my PATH was not sufficient.  You also need to have `etcd` and `kube-apiserver` binaries available, and afaict you have to specify where they are using the KUBEBUILDER_ASSETS env var.

**Review by kylape — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (bsquizz) [reply README.md:85]: True, when I installed kubebuilder, those two were included. Once they are on the PATH it appears setting the ASSETS is not needed for me.

Comment (bsquizz) [reply README.md]: Updated to mention the other executables

Comment (bsquizz) [reply README.md]: As long as kubebuilder is installed in the default location (/usr/local/kubebuilder/bin) you don't need KUBEBUILDER_ASSETS ... if you place those in a different dir, you do need that env var. Updated README.

**Pull request merged**